clippy_lints/src/forced_return.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014-2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::rustc::hir::{intravisit::FnKind, Body, ExprKind, FnDecl};
+use crate::rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use crate::rustc::{declare_tool_lint, lint_array};
+use crate::rustc_errors::Applicability;
+use crate::syntax::{ast::NodeId, source_map::Span};
+use crate::utils::{snippet_opt, span_lint_and_then};
+
+/// **What it does:** Checks for missing return statements at the end of a block.
+///
+/// **Why is this bad?** Actually it is idiomatic Rust code. Programmers coming
+/// from other languages might prefer the expressiveness of `return`.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// fn foo(x: usize) {
+///     x
+/// }
+/// ```
+/// add return
+/// ```rust
+/// fn foo(x: usize) {
+///     return x;
+/// }
+/// ```
+declare_clippy_lint! {
+    pub FORCED_RETURN,
```

Comment:
I'm not very happy with the name. Just trying to come up with a name I keep running into "those returns are so very unidiomatic".

At the very least it must be `forced_returns` so that `allow(forced_returns)` reads like a sentence.

I'm gonna have to reread the Clippy 1.0 RFC to figure out whether we can accept this lint, too. (Yes, I realize I commented on the issue with a positive response)

Reply:
How about `missing_returns` as it should be stating the thing the lint checks for?

Reply:
maybe `implicit_trailing_returns`?

Reply:
I changed it to missing_returns for now (https://github.com/rust-lang/rust-clippy/pull/3494/commits/978f8c65ee44b43243a83a047be6cdacb6df1320), makes more sense to me at least.

Reply:
Renamed it to `implicit_return` accordign to @Manishearth in https://github.com/rust-lang/rust-clippy/pull/3494/commits/aed2b986e6d596f6ed154625030b8c83a67066fe.